reVX/turbine_flicker/turbine_flicker.py

```diff
@@ -0,0 +1,563 @@
+        out = MeanWindDirectionsPoint.run(
+            gid,
+            excl_fpath,
+            res_fpath,
+            tm_dset,
+            wind_dir_dset,
+            resolution=resolution,
+            exclusion_shape=exclusion_shape)
+
+        meta = out['meta']
+        wind_dir = out[wind_dir_dset]
+
+        # Drop last day of leap years
+        if len(wind_dir) == 8784:
+            wind_dir = wind_dir[:-24]
+
+        return meta, wind_dir
+
+    @classmethod
+    def _compute_shadow_flicker(cls, lat, lon, blade_length, wind_dir):
+        """
+        Compute shadow flicker for given location
+
+        Parameters
+        ----------
+        lat : float
+            Latitude coordinate of turbine
+        lon : float
+            Longitude coordinate of turbine
+        blade_length : float
+            Turbine blade length. Hub height = 2.5 * blade length
+        wind_dir : ndarray
+            Time-series of wind direction for turbine
+
+        Returns
+        -------
+        shadow_flicker : ndarray
+            2D array centered on the turbine with the number of flicker hours
+            per "exclusion" pixel
+        """
+        mult = (cls.FLICKER_ARRAY_LEN * cls.GRIDCELL_SIZE) / 2
+        mult = mult / (blade_length * 2)
+        FlickerMismatch.diam_mult_nwe = mult
+        FlickerMismatch.diam_mult_s = mult
+        FlickerMismatch.steps_per_hour = cls.STEPS_PER_HOUR
+        FlickerMismatch.turbine_tower_shadow = False
+
+        assert len(wind_dir) == 8760
+
+        shadow_flicker = FlickerMismatch(lat, lon,
+                                         blade_length=blade_length,
+                                         angles_per_step=None,
+                                         wind_dir=wind_dir,
+                                         gridcell_height=cls.GRIDCELL_SIZE,
+                                         gridcell_width=cls.GRIDCELL_SIZE,
+                                         gridcells_per_string=1)
+        shadow_flicker = shadow_flicker.create_heat_maps(range(0, 8760),
+                                                         ("time", ))[0]
+
+        return shadow_flicker
+
+    @staticmethod
+    def _check_shadow_flicker_arr(shadow_flicker):
+        """
+        Check to ensure the shadow_flicker array is odd in shape, i.e. both
+        dimensions are odd allowing for a central pixel for the turbine to
+        sit on. Flip 0-axis to mimic the turbine sitting on each building.
+        All flicker pixels will now indicate locations where a turbine would
+        need to be to cause flicker on said building
+
+        Parameters
+        ----------
+        shadow_flicker : ndarray
+            2D array centered on the turbine with the number of flicker hours
+            per "exclusion" pixel
+
+        Returns
+        -------
+        shadow_flicker : ndarray
+            Updated 2D shadow flicker array with odd dimensions if needed
```

Comment:
And also inverted right? seems important that someone should not expect basically the same outputs as input.